### ISSUE TITLE
Reference links require protocol prefix

### ISSUE THREAD

**Issue by WovenTales** [lervag/wiki.vim#275]
# Description

I've been wanting to switch to reference links for a while, given how Vim's concealment interacts with word wrapping, but hadn't had any luck getting it to work.  Reading through another issue here for a second or third time, I finally noticed that they prefixed their example path with `wiki:` -- and when I added that to mine, everything finally navigated properly.  I had never considered that as being the issue since inline links *don't* require a protocol to be specified.  Now that I'm aware of the limitation, I can certainly work around it, but I wonder if it wouldn't be better to allow in-wiki reference links to just name the bare path identically to in-wiki in-line links; or if there's a technical or format-based reason to enforce the difference, then the documentation should make the requirement clear (there's nothing in `:help wiki-link-reference` about the protocol either way).

# Minimal working example

```markdown
* This is a functional link: [link](wiki:target.md).
* This is as well: [link](target.md).
* This link works: [link][valid].
* This one doesn't: [link][bad].

  [valid]: wiki:target.md
  [bad]: target.md
```

**Comment by lervag** [comment]
I don't see any good reasons not to allow the "unprotocolled" urls for reference links. I'm making it with one minor caveat, though: If the target does not exist, then we will instead move the cursor to the recognized reference (similar to the current behaviour).

Let me know what you think!

**Comment by WovenTales** [comment]
Yeah, that looks good, thanks!  Jumping down to the reference is intuitive enough, and I can certainly see how it may avoid unexpected behaviour.  It's still a bit weird to jump down, `W` over to the link and hit enter only to have the bare path be wrapped in a link syntax rather than opening the expected page, but between that and potentially linking via something that was only intended as a footnote, I definitely appreciate the current behaviour.
